src/transforms/groupby.js

```diff
@@ -128,9 +155,20 @@ function transformOne(trace, state) {
 
     var arrayAttrs = PlotSchema.findArrayAttributes(trace);
 
-    var style = opts.style || {};
+    var style = opts.style || [];
+    var styleIndex = style;
+
+    // If the styles are an array, then translate it into an index keyed on the
+    // target. Otherwise just use the object provided for the sake of backward
+    // compatibility.
+    if(Array.isArray(style)) {
```

Comment:
Note: I could have translated the object in to array style format to get a nice consistent result on `fullData`, except the useful format is actually as originally defined since it needs a lookup so that it would just have to be translated *back* to the key-value style down below. That way would be a bit cleaner and may need to happen; this way just minimizes the back and forth. No strong feelings here.